main.go

```diff
@@ -329,3 +309,72 @@ func filterTimetableBySubjects(t timetable.Timetable, codes []string) timetable.
 	}
 	return filtered
 }
+
+var (
+	subjectscacheExpirationTime = time.Hour * 4
+	subjectscache 				= cache.New(subjectscacheExpirationTime, time.Hour*6)
+)
+
+// The return type is a map that for every year of the course map a curriculum
+// to a slice of subjects
+func getSubjectsMapFromCourseAndCurricula(course *unibo_integ.Course, curricula map[int]curriculum.Curricula) (map[int]map[curriculum.Curriculum][]timetable.SimpleSubject, error) {
+	if course == nil {
+		return nil, fmt.Errorf("course parameter is nil")
+	}
+
+	if curricula == nil {
+		return nil, fmt.Errorf("curricula parameter is nil")
+	}
```

Comment:
Queste due condizioni sono invarianti, non dovrebbero dare errore. Al massimo puoi fare `panic()`, ma comunque verrebbe fatto alla prima de-referenziazione dei puntatori. Toglierei questi due controlli.

Reply:
Riguardo alla variabile `course` dovrebbe essere sempre definita e mai nil, mentre `curricula` potrebbe essere nil perché li richiediamo da unibo e se in quel momento non risponde per qualche motivo allora diventa nil.

Curricula potrebbe quindi diventare nil anche dopo molto tempo che il programma sta girando. Sinceramente più che mandare tutto in panic farei un piccolo check in più con un if e lascerei il codice così

Reply:
Va bene, ma aggiungi un commento perché la voglia di toglierle è tanta xD